contrib/meson/meson/meson.build

```diff
@@ -8,105 +9,60 @@
 # #############################################################################
 
 cc = meson.get_compiler('c')
+
 pkgconfig = import('pkgconfig')
-c_std = get_option('c_std')
-default_library = get_option('default_library')
+fs = import('fs')
 
-host_machine_os = host_machine.system()
-os_windows = 'windows'
-os_linux = 'linux'
-os_darwin = 'darwin'
-os_freebsd = 'freebsd'
-os_sun = 'sunos'
+lz4_source_root = fs.parent(fs.parent(fs.parent(meson.current_source_dir())))
```

Comment:
You cannot use fs.parent, because the fs module is not available until meson 0.53, but before Meson 0.61 we forgot to add a FeatureNew warning about it.

However it is not strictly needed. Instead of every file using `../../../` with different levels, you can define it once, here, but using `../` instead of `fs.parent()`

`meson.current_source_dir() / '../../../'`

Reply:
I get the following error:

```
../meson/lib/meson.build:39:0: ERROR: Tried to form an absolute path to a source dir.
You should not do that but use relative paths instead.
```

Reply:
For now I just bumped to minimum Meson version to 0.53 as mentioned.

Reply:
(I didn't test it, sorry.)

Oh right... it should be an absolute path in both cases.

The difference is that, given rootdir: `/tmp/lz4/contrib/meson`:
- in the `fs.parent(rootdir)` case it is: `/tmp/lz4/lib`
- in the `currentdir / '../../../'` case it is: `/tmp/lz4/contrib/meson/meson/../../../lib`

When integrating lz4 into the wrapdb, we move the meson.build into the top level of the lz4 source tree so `/tmp/lz4` will be the rootdir and both methods will spawn an error.

Sorry, but you cannot simplify this. Please continue to hardcode a set number of `../../../` in the header of each file. It is necessary as long as the Meson files are in `contrib/meson/` instead of in `.` and there is no getting around it (well, other than putting them back where Meson expects them to be).